CIP-1694/README.md

```diff
@@ -0,0 +1,624 @@
+
+#### Replacing the Constitutional Committee
+
+The Constitutional Committee can be replaced in one of two ways:
+
+* When in a **normal** state (i.e. a state of **confidence**), the committee can be replaced via a specific governance action (action 2 below) that requires the approval of both the **current Constitutional Committee** and the **DReps**.
+
+* When in a state of **no-confidence**, the committee can also be replaced via a specific governance action (action 5 below), but this requires the approval of the **SPOs** and the **DReps** instead.
+
+#### Size of the Constitutional Committee
+
+Unlike the Shelley governance design, the size of the Constitutional Committee is not fixed.  It may be changed any time that a new committee is installed.  Likewise, the _quorum_ (the number of votes that are required to enact governance actions) is not fixed and can be varied whenever a new committee is installed. This gives a great deal pf flexibility.
+
+### Governance Actions
+
+We define six different types of **governance actions**. A governance action is an on-chain event that is triggered by a transaction and has a deadline after which it cannot be enacted.
+
+An action is said to be **ratified** when it gathers enough votes in its favor (through rules and parameters detailed below). An action that doesn't collect sufficient `yes` votes before its deadline is said to have **expired**.
+An action that has been ratified is said to be **enacted** once it has been activated on the network.
+Regardless of whether they have been ratified, actions may, however, be **dropped** without being **enacted** if, for example, a motion of no confidence is enacted.
+
+
+| Action                                         | Description                                                                                        |
+| ---                                            | ---                                                                                                |
+| 1. Motion of no-confidence                     | A motion to create a _state of no-confidence_ in the current Constitutional Committee  |
+| 2. New Constitutional Committee and/or quorum size | Changes to the members of the Constitutional Committee and/or to its signature threshold   |
+| 3. Updates to the Constitution                 | A modification to the off-chain Constitution, recorded as an on-chain hash of the text document                          |
+| 4. Hard-Fork[^2] Initiation                    | Triggers a non-backwards compatible upgrade of the network; requires a prior software upgrade |
```

Comment:
Can you elaborate on `requires a prior software upgrade`? Is this going to be enforceable on chain where a governance proposal can't even be submitted unless some number of stake pools have upgraded their node version (averaged by stake delegated).

Reply:
When we want new features, new code has to be written to support the new features, and so SPOs need to upgrade the software. It's fine for the action to precede the SPOs upgrading, we just do not want to enact it before everyone is ready, which is precisely why the SPOs should vote. There is no way to know for sure who has upgraded, though perhaps the idea in CIP-47 could be incorporated into this CIP (though I think it is probably a lot of complication for little gain).